src/types/st-object.ts

```diff
@@ -67,6 +67,9 @@ const STObject = makeClass(
           return;
         }
         serializer.writeFieldAndValue(field, value);
+        if (field.type.name === "STObject") {
+          serializer.put(Buffer.from([0xe1]));
```

Comment:
Note: `0xe1` won't stay as a magic number, it will be refactored with STObject.

Reply:
Oh, I see, you've moved this here, hence metadata working